tests/e2e/volumeproperty_test.go

```diff
@@ -0,0 +1,175 @@
+7. Again verify capacity and disk format using docker cli and admin cli to make sure things are fine after running the container.
+
+NOTE: Do steps 5 ,6 and 7 only for volume of 'thin' disk format
+*/
+
+func (s *VolumePropVerificationSuite) TestVolumeProperties(c *C) {
+	log.Printf("START: TestVolumeProperties")
+
+	// creating volumes of all disk formats
+	s.createVolumes(c)
+	for _, volumeName := range s.volumeNames {
+		// getting and verifying the admin cli and docker cli values
+		s.getPropertiesAndVerify(c, volumeName)
+		s.dockerVolumeRemoveCmd = s.dockerVolumeRemoveCmd + " " + volumeName
+	}
+
+	// attaching only thin volume and verifying its properties
+	out, err := dockercli.AttachVolume(s.dockerHostIP, s.volumeNames[0], s.containerName)
+	c.Assert(err, IsNil, Commentf("Failed to attach the volume [%s]", out))
+	s.vmAttached = true
+	s.getPropertiesAndVerify(c, s.volumeNames[0])
+
+	log.Printf("END: TestVolumeProperties")
+}
+
+// createVolumes - creates volumes of each format type
+func (s *VolumePropVerificationSuite) createVolumes(c *C) {
+	for i, formatType := range s.formatTypes {
+		s.containerName = inputparams.GetContainerNameWithTimeStamp(TestName)
+		s.volumeNames[i] = inputparams.GetVolumeNameWithTimeStamp(TestName)
+		out, err := ssh.InvokeCommand(s.dockerHostIP, dockerconst.CreateVolume+"--name="+s.volumeNames[i]+" -o diskformat="+formatType)
+		c.Assert(err, IsNil, Commentf(out))
+	}
+}
+
+// getPropertiesAndVerify - gets properties of a volume from esx and docker host
+func (s *VolumePropVerificationSuite) getPropertiesAndVerify(c *C, volumeName string) {
+	valuesAdminCli, err := verification.GetVolumePropertiesAdminCli(volumeName, s.esxIP)
+	c.Assert(err, IsNil, Commentf(valuesAdminCli))
+
+	valuesDockerCli, err := verification.GetVolumePropertiesDockerCli(volumeName, s.dockerHostIP)
+	c.Assert(err, IsNil, Commentf(valuesDockerCli))
+	s.verifyProperties(c, volumeName, valuesAdminCli, valuesDockerCli)
+}
+
+// verifyProperties - verifies admin cli and docker cli values for size, disk format and attached-to-vm field.
+func (s *VolumePropVerificationSuite) verifyProperties(c *C, volumeName, valuesAdmincli, valuesDockerCli string) {
+	log.Printf("Verifying volume properties like size, disk-format and attached-to-vm fields "+
+		"at vm [%s] and esx [%s] for volume - [%s] ", s.dockerHostIP, s.esxIP, volumeName)
+	arrAdminCliValues := strings.Fields(valuesAdmincli)
+	sliceAdminCliValues := arrAdminCliValues[1:len(arrAdminCliValues)]
+	// status := verification.VerifyAttachedStatus(volumeName, s.dockerHostIP, s.esxIP)
+
+	sliceDockerCliValues := s.arrangeDockerCliValues(valuesDockerCli)
+	for i := range sliceDockerCliValues {
+		c.Assert(len(sliceDockerCliValues), Equals, len(sliceAdminCliValues), Commentf("Values mis-match between admin cli and docker cli for volume %s", volumeName))
+		if i == 2 && !s.vmAttached {
```

Comment:
what is so special about `i==2`? you may want to put some inline comment to make reviewer's life easier.